hashdist/cli/manage_store_cli.py

```diff
@@ -114,3 +114,137 @@ def run(ctx, args):
+        from pcs_api.storage import StorageFacade
+        from pcs_api.bytes_io import (MemoryByteSource, MemoryByteSink,
+                                      FileByteSource, FileByteSink,
+                                      StdoutProgressListener)
+        from pcs_api.models import CPath, CFolder, CBlob, CUploadRequest, CDownloadRequest
+        if args.dryrun:
+            from ..core import BuildStore
+            store = BuildStore.create_from_config(ctx.get_config(), ctx.logger)
+            os.chdir(store.artifact_root)
+            for package in os.listdir(store.artifact_root):
+                for artifact in os.listdir(pjoin(store.artifact_root,package)):
+                    print "tar czvf {package}/{artifact}.tar.gz {package}/{artifact}".format(package=package,artifact=artifact)
+            import pdb
+            pdb.set_trace()
+            os.system('cd %s; tar xzvf %s; rm -f %s' % (self.artifact_root,temp_path,temp_path))
+        else:
+            import pdb
+            pdb.set_trace()
+            remote_path = pjoin(DEFAULT_STORE_DIR,"remote")
+            app_info_path=pjoin(remote_path,"app_info_data.txt")
+            user_credentials_path=pjoin(remote_path,"user_credentials_data.txt")
+            if not os.path.exists(app_info_path):
+                print('Not found any application information repository file: ', app_info_path)
+                print('Refer to documentation and class AppInfoFileRepository to setup pcs_api for a quick test')
+                exit(1)
+            apps_repo = AppInfoFileRepository(app_info_path)
+            if not os.path.exists(user_credentials_path):
+                print('Not found any users credentials repository file: ', user_credentials_path)
+                print('Refer to documentation and class UserCredentialsFileRepository to setup pcs_api for a quick test')
+                exit(1)
+            user_credentials_repo = UserCredentialsFileRepository(user_credentials_path)
+            provider_name = apps_repo._app_info.keys()[0].split(".")[0]
+            app_info = apps_repo.get(provider_name)
+            user_info = user_credentials_repo.get(app_info)
+            storage = StorageFacade.for_provider(provider_name) \
+                       .app_info_repository(apps_repo,app_info.app_name) \
+                       .user_credentials_repository(user_credentials_repo,user_info.user_id) \
+                       .build()
+            print("user_id = ", storage.get_user_id())
+            print("quota = ", storage.get_quota())
+            from ..core import BuildStore
+            store = BuildStore.create_from_config(ctx.get_config(), ctx.logger)
+            os.chdir(store.artifact_root)
+            for package in os.listdir(store.artifact_root):
+                for artifact in os.listdir(pjoin(store.artifact_root,package)):
+                    os.system("tar czvf {package}/{artifact}.tar.gz {package}/{artifact}".format(package=package,artifact=artifact))
```

Comment:
Instead of `os.system`, use `subprocess.check_call`, which will raise an exception if the call fails and won't fail if there's a space in `{artifact}` or `package` (which I guess there'll never be, but check_call is just always better).